### DISCUSSION TITLE
ffmpeg results in low bitrate and no webrtc

### DISCUSSION THREAD

**Discussion by madjxatw** [bluenviron/mediamtx#1734] [Questions]
Hi, I've attempted to use both gstreamer and ffmpeg on Linux to capture a h264 webcam and publish to RTSP server, but ffmpeg resulted in a very low bitrate  (around 1000kbps) at client side while gstreamer is at around 10000kbps. In addition, I didn't get webrtc stream while using ffmpeg, with a message stating "closed (the stream doesn't contain any supported codec (which are currently VP9, VP8, H264, Opus, G722, G711))"

ffmpeg command:

```sh
ffmpeg -f v4l2 -i /dev/video2 -video_size 1920x1080 -framerate 25 -f rtsp rtsp://<ip>:<port>/cam
```

gstreamer command:

```sh
gst-launch-1.0 -v v4l2src device=/dev/video2 ! video/x-h264,width=1920,height=1080,framerate=25/1 ! rtspclientsink location=rtsp://<ip>:<port>/cam
```

Any advice to make ffmpeg work correctly?

**Comment by aler9** [comment]
Hello, this probably happens because FFmpeg chose the wrong video format. List available formats for your camera:

```
ffmpeg -f v4l2 -list_formats all -i /dev/video2
```

If H264 is listed among the native formats, you can choose it with input_format:

```
ffmpeg -f v4l2 -i /dev/video2 -input_format h264 -video_size 1920x1080 -framerate 25 -f rtsp rtsp://<ip>:<port>/cam
```

[a comment by an automated account is not shown]
